Bottom: c5bdddfe8b2eceb6b97cf4d8ceb949a2db089c2a
Top:    e9d0c256bc7df3b9e0fe71c3f6af29a500fbbfd0
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-23 17:55:22 +0000

More hack to avoid GL calls when we're out of context


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 76095fe..2d1e7db 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -505,6 +505,12 @@ ghid_set_color (hidGC gc, const char *name)
 
   gc->colorname = (char *) name;
 
+  if (!check_gl_drawing_ok_hack)
+    {
+      current_color = NULL;
+      return;
+    }
+
   if (gport->colormap == NULL)
     gport->colormap = gtk_widget_get_colormap (gport->top_window);
   if (strcmp (name, "erase") == 0)
